srttools/monitor/webserver.py

```diff
@@ -3,233 +3,51 @@
-                img2.src = images[rightIndex];
-
-                let feed_id = document.createElement("div");
-                feed_id.classList.add("feed_id");
-                feed_id.id = "feed-" + pairIndex;
-
-                div.appendChild(feed_id);
-                div.appendChild(img1);
-                div.appendChild(img2);
-                document.getElementById("thumbnail-bar").appendChild(div);
-            }
-        }
-
-        function updateMainView() {
-            let mainPair = changePairsVisibility();
-
-            if (mainPair !== """
-        + str(MAX_FEEDS)
-        + """) {
-                let leftIndex = mainPair * 2;
-                let rightIndex = leftIndex + 1;
-                document.getElementById("main-left").src = images[leftIndex];
-                document.getElementById("main-right").src = images[rightIndex];
-            } else {
-                document.getElementById("main-left").src = whiteImage;
-                document.getElementById("main-right").src = whiteImage;
-            }
-        }
-
-        function addImage(index, base64) {
-            images[index] = base64 ? "data:image/png;base64," + base64 : whiteImage;
-
-            let pairIndex = Math.floor(index / 2);
-            let leftIndex = pairIndex * 2;
-            let rightIndex = leftIndex + 1;
-
-            addPair(pairIndex);
-
-            document.getElementById("thumb-" + index).src = images[index];
-
-            updateMainView();
-        }
-
-        function connect() {
-            let ws = new WebSocket("ws://" + destination + ":"""
-        + str(port)
-        + """/images");
+    raise ImportError(
+        "To use SDTmonitor, you need to install tornado: \n" "\n   > pip install tornado"
+    )
 
-            ws.onmessage = function(message) {
-                let msg = JSON.parse(message.data);
-                addImage(msg.index, msg.image);
-            };
+from srttools.monitor.common import log
 
-            ws.onclose = function() {
-                setTimeout(connect, 10000);
-            };
-        }
 
-        connect();
-    </script>
-</body>
-</html>"""
-    )
-    with open("index.html", "w") as fobj:
-        print(html_string, file=fobj)
+def create_index_file(**kwargs):
+    config = {
+        "port": 8080,
+        "reconnectTime": 10000,
+        "maxScrollSpeed": 80,
+    }
+    config.update(kwargs)
+    with open("index.html", "w") as webpage:
+        with open(files("srttools.monitor").joinpath("resources", "index.html"), "r") as template:
+            print(Template(template.read()).safe_substitute(config), file=webpage, end="")
 
 
 class WSHandler(WebSocketHandler):
+    # This allows clients that did not send any request to the HTTPHandler previously
+    # i.e.: a client that opens the index.html page instead of accessing it via network
+    check_origin = lambda _, __: True
+    # We don't care about listening to messages
+    on_message = lambda _, __: None
```

Comment:
Simple trick to avoid defining multi-line methods. This trick also improve coverage ;)